### ISSUE TITLE
Ore Generation

### ISSUE THREAD

**Issue by Wolverine903** [RebelKeithy/Metallurgy3#2]
Maybe I'm doing something wrong, but no ores generate in my worlds i make. I can access the Metallurgy stuff when I change my mode to creative, but still no ores are in the world. Even the vanilla ores are missing. Is there some setting I need to change or is this a bug?

**Comment by Vyrebird** [comment]
I see the same - no Metallurgy (or vanilla) ores in the Overworld, but the Nether ores seem to work (vanilla quartz, and  Metallurgy Ignatius, Shadow Iron, Lemurite, and Vyroxeres, at least).
Other mod's ores (eg Forestry's copper/tin) continue to generate fine in the overworld.

(I'm also a newbie to Metallurgy, so there could be some config setting we're both simply missing)

**Comment by RebelKeithy** [comment]
No, it should be generating, are you sure they're not generating at all, or are they just really scarce. I'll check and see if I can find something wrong.

**Comment by Wolverine903** [comment]
I am fairly positive they are not generating, I went through a part of a cave and did not see any ores.  I dug numerous vertical tunnels in creative mode down to bedrock and never saw any of the ores either.

**Comment by Wolverine903** [comment]
Alright I tried another world and explored a huge cave for the last 10 minutes, the only ore I found was 2 blocks of emerald ore, each one in a different spot.  So it looks like only emerald is spawning.

**Comment by Vyrebird** [comment]
I've made a few shafts in Creative by hand, and about a dozen more with TNT, boring from the surface to bedrock, as I experimented with MetallurgyCore.cfg (to no avail).  I only saw Forestry ores in that time (in retrospect, I should've uninstalled Forestry to keep the test 'pure', but they were handy to confirm that I _should_ be seeing ores).  None of my shafts were in Extreme Hills (Edge), so I can't comment on Emerald Ore.  I also tried bumping up the frequency of Platinum (vein size 50, veins per chunk 20) without any change.

I made two similar TNT shafts in a 1.4.7 world (with Metallurgy2), and encountered multiple Metallurgy ores in both of them, as a further test of the technique's validity.

**Comment by RebelKeithy** [comment]
ok, thanks, definitely something wrong then. This update could possibly have fixed it, try it and let me know.

**Comment by Vyrebird** [comment]
Problem remains in 3.0.0.0.2, seemingly unchanged (have sighted more Nether ores, but that's probably just from doing more looking).  I'm guessing the console spam of

> 2013-03-21 02:21:04 [INFO] [STDOUT] checking dim '0' against0
> 2013-03-21 02:21:04 [INFO] [STDOUT] checking dim '-1' against0

is related debugging?

I can confirm Emerald Ore was indeed spawning in 3.0.0.0.1, btw.

It's probably unrelated, but the config parameter is spelled "Diminsions", whereas the proper spelling is "Dimensions".

**Comment by RebelKeithy** [comment]
I'm looking into it, I have found that vanilla ores aren't spawning and I'm in the process of fixing that.

The debug output should have been removed for the beta, I just forgot to comment it out.

**Comment by Vyrebird** [comment]
3.0.0.0.3 seems to have returned the vanilla ores - iron, gold, and diamond as indicated by the debug stick (took me a minute to find, I hadn't realized how literal you were being :p), as well as coal, redstone, and lapis.  They appear to ignore the config file.

As in 3.0.0.0.1/2, the Nether ores also continue to spawn, and obey the config file (at least for simple frequency changes).

**Comment by RebelKeithy** [comment]
I believe I have fixed the problem in 3.0.0.0.4, try it and let me know what you find

**Comment by Wolverine903** [comment]
Looks great, the ores are back!  Got a question though, maybe it is suppose to be like this, but the fantasy ores seem to sparkle to much or the sparkles are too large and are very noticeable from a distance.  I don't quite remember them like that in Metallurgy 2.  Hopefully the attached images show up for you to see.

![2013-03-22_01 29 00](https://f.cloud.github.com/assets/3899022/289512/d9b8320e-92b1-11e2-876b-57f318ecf0bc.png)

![2013-03-22_01 27 02](https://f.cloud.github.com/assets/3899022/289517/0cb0872e-92b2-11e2-974b-f4a435714f8a.png)

**Comment by Vyrebird** [comment]
3.0.0.0.4 looks good to me.  The overworld ores seem to be spawning (didn't explicitly check all of them, but most).  Quick checks indicate they respond to the config files (modifying platinum's height & size, changing Alduorite to an overworld-only ore with high frequency & size).  So, looks quite resolved to me :)

If you don't mind me asking, what did the cause wind up being?  It seemed so puzzling to me that the Nether ores would work, but none of the rest would.

As for the fantasy ores particles bit, I think it's looking for the sparkle texture in the wrong place, considering this from the console log:

> 2013-03-22 19:22:14 [INFO] [Minecraft-Client] [CHAT] Your game mode has been updated
> 2013-03-22 19:22:36 [INFO] [STDERR] java.io.FileNotFoundException: /Metallurgy/FantasyMetalsParticle.png
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.texturepacks.TexturePackDefault.func_98139_b(SourceFile:30)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.texturepacks.TexturePackImplementation.func_98137_a(SourceFile:82)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.texturepacks.TexturePackImplementation.func_77532_a(SourceFile:93)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.renderer.RenderEngine.func_78341_b(RenderEngine.java:208)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at rebelkeithy.mods.metallurgy.metals.EntityFantasyOreFX.func_70539_a(EntityFantasyOreFX.java:55)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.particle.EffectRenderer.func_78874_a(EffectRenderer.java:121)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1158)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:991)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:867)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at net.minecraft.client.Minecraft.run(Minecraft.java:756)
> 2013-03-22 19:22:36 [INFO] [STDERR]   at java.lang.Thread.run(Unknown Source)

The texture is actually at mods\Metallurgy\textures\particles\FantasyMetalsParticle.png .  Moving the file to Metallurgy\FantasyMetalsParticle.png seems to fix the issue:
![fantasy_ore_particles_modified](https://f.cloud.github.com/assets/2203959/293202/33107c46-934b-11e2-810a-d92dd72b3b90.png)

**Comment by RebelKeithy** [comment]
The cause was isMineGenReplaceable changed, before it returned true if the block was stone, now you passed in an id and it returned true if it the block id was the same as the given id. The reason it worked for nether ores is that I was using isMineGenReplaceable for stone and simply a direct comparison of ids for nether rack.

As for the fantasy particles, I moved the images around trying to get them organized, but forgot to change the code. I'll get that fixed for 3.0.0.0.5 (which may end up being 3.0.1). But I've got several exams coming up, so I'm not sure when that will be.
